pygmt/datasets/earth_mean_dynamic_topography.py

```diff
@@ -0,0 +1,104 @@
+"""
+Function to download the CNES Earth mean dynamic topography dataset from the GMT data
+server, and load as :class:`xarray.DataArray`.
+
+The grids are available in various resolutions.
+"""
+
+from collections.abc import Sequence
+from typing import Literal
+
+import xarray as xr
+from pygmt.datasets.load_remote_dataset import _load_remote_dataset
+
+__doctest_skip__ = ["load_earth_mean_dynamic_topography"]
+
+
+def load_earth_mean_dynamic_topography(
+    resolution: Literal["01d", "30m", "20m", "15m", "10m", "07m"] = "01d",
```

Comment:
I think we need to add a note that `07m` acutally points to a grid with 7.5 arc-minute based on https://github.com/GenericMappingTools/gmtserver-admin/blob/e16841a7260b2c8d075eddc9def69e59569f0cf7/recipes/earth_mdt.recipe#L35.

Actually, we also need to add similar notes to some other datasets like mars_relief, mercury_relief, moon_relief, and pluto_relief.

```
$ grep 'master$' * | awk 'NF==5'
earth_age.recipe:01		m		30		4096	master
earth_dist.recipe:01		m		30		4096	master
earth_edefl.recipe:01		m		30		4096	master
earth_faa.recipe:01		m		30		4096	master
earth_faaerror.recipe:01		m		30		4096	master
earth_gebco.recipe:15		s		10		4096	master
earth_gebcosi.recipe:15		s		10		4096	master
earth_geoid.recipe:01		m		30		4096	master
earth_mag.recipe:02		m		60		4096	master
earth_mag4km.recipe:02		m		60		4096	master
earth_mdt.recipe:7.5		m		0		4096	master
earth_mss.recipe:01		m		30		4096	master
earth_ndefl.recipe:01		m		30		4096	master
earth_relief.recipe:15		s		10		4096	master
earth_synbath.recipe:15		s		10		4096	master
earth_vgg.recipe:01		m		30		4096	master
earth_wdmam.recipe:03		m		90		2048	master
mars_relief.recipe:12.1468873601	s		25.7142857143		4096	master
mercury_relief.recipe:56.25	s		30		4096	master
moon_relief.recipe:14.0625	s		10		4096	master
pluto_relief.recipe:52.0732883317		s		30		4096	master
venus_relief.recipe:01		m		30		4096	master
```

Reply:
Nice catch!
I added a note to the docstring for the `resolution` parameter in commit https://github.com/GenericMappingTools/pygmt/pull/3718/commits/af6cf7aecf7eaa692e970758c48a2249b412dfd1 and submitted PR #3721 for the other four datasets.